fix: Handling ip addresses in format of ::ffff:192.168.0.1

John222 opened the pull request:
In case the haproxy is running in Ipv4 and IPv6 mode, the haproxy maps Ipv4 addresses in the IPv6 format. This fix converts the IPv4 mapped address in IPv6 format back to a regular IPv4 address like 192.168.0.1

John222 committed:
fix: Handling ip addresses in format of ::ffff:192.168.0.1

In case the haproxy is running in Ipv4 and IPv6 mode, the haproxy maps Ipv4 addresses in the IPv6 format. This fix converts the IPv4 mapped address in IPv6 format back to a regular IPv4 address like 192.168.0.1

superstes commented:
@John222 Yeah - makes total sense.
Do you mind trying a non-regex alternative? It might perform marginally better

```lua
local V4_MAPPED_PREFIX = "::ffff:"
local PREFIX_LEN = 7

local function normalize_ip(ip)
    if ip:sub(1, PREFIX_LEN) == V4_MAPPED_PREFIX then
        return ip:sub(PREFIX_LEN + 1)
    end
    return ip
end
```

In my tests it worked

John222 commented:
Hi @superstes i tried your suggestion. It worked.

John222 committed:
Fixing issue that go backend sends strings with quotes

Fixing the issue that the go backend is returning the string with quotes which breaks in haproxy when comparing the string. As haproxy escapes quotes to '#22' ascii character.

John222 committed:
Fixing wrong changes

Fixing typo
